### PR TITLE
Fixes to JIRAs 1793,1807,1808. Details below.

### PR DESCRIPTION
**Pull request by anoopsharma00**
1807: timezone specification support in datetime values
1808: query cache bug fix with time/timestamp values with fraction
1793: get stmt with views now include schema names

### CODE REVIEW

File: core/sql/common/DateTimeType.cpp [hunk 1]
```diff
@@ -171,6 +172,8 @@ NABoolean DatetimeType::isEncodingNeeded() const
 
 NABoolean DatetimeType::operator==(const NAType& other) const
 {
+  DatetimeType &otherDT = (DatetimeType&)other;
```
Comment (DaveBirdsall) [inline core/sql/common/DateTimeType.cpp]: Not sure why this variable is defined. We don't use it anywhere. Doesn't hurt anything though.

File: core/sql/common/DateTimeType.cpp [hunk 2]
```diff
@@ -1399,18 +1403,69 @@ NABoolean SQLMPDatetime::isCompatible(const NAType& other, UInt32 * flags) const
...
+  flags |= CONV_NO_HADOOP_DATE_FIX;
+  short rc = 
+    ExpDatetime::convAsciiToDatetime((char*)strValue, strlen(strValue), 
+                                     dstValue, 100,
+                                     startField, endField, 
+                                     ExpDatetime::DATETIME_FORMAT_NONE,
+                                     trueFP,
+                                     NULL, &diags, flags);
+  if (rc)
+    return;
```
Comment (DaveBirdsall) [inline core/sql/common/DateTimeType.cpp]: fractionPrecision will be uninitialized if we take this return

**Comment by DaveBirdsall** [comment]
Looks good to me, apart from one error path that might be a concern.

**Comment by DaveBirdsall** [comment]
jenkins, add user

File: core/sql/common/DateTimeType.cpp (hunk 2 above)
Comment (anoopsharma00) [inline core/sql/common/DateTimeType.cpp]: That return is for error paths. The Datetime value being constructed will be invalid and
not be used. isValid() method will return true.
But we can initialize fractionPrecision to zero on entry to this method.

Comment (anoopsharma00) [inline core/sql/common/DateTimeType.cpp]: correction: isValid() method will return FALSE on error path.

**Pull request merged**